android-agent/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -402,27 +387,17 @@ private void applyConfiguration() {
         // Network specific attributes
         if (config.shouldIncludeNetworkAttributes()) {
             // Add span processor that appends network attributes.
-            CurrentNetworkProvider currentNetworkProvider = getOrCreateCurrentNetworkProvider();
             addTracerProviderCustomizer(
                     (tracerProviderBuilder, app) -> {
                         SpanProcessor networkAttributesSpanAppender =
-                                NetworkAttributesSpanAppender.create(currentNetworkProvider);
+                                NetworkAttributesSpanAppender.create(
+                                        ServiceManager.get().getCurrentNetworkProvider());
```

Comment:
Does this also eventually move to the network instrumentation module? It currently strange to me because the main agent is aware that some network attributes exist, and it is including them based on config, but there's no 
clear indication about where these network attributes originate (hint: it's in a separate instrumentation module).

Reply:
This feature seems to be only for adding global attrs with a SpanProcessor, so I think it should live in the core module to be available before initialization. I've added more details on it [in this other comment](https://github.com/open-telemetry/opentelemetry-android/pull/408#discussion_r1625825393).